src/udiskslinuxfilesystem.c

```diff
@@ -1913,7 +1913,7 @@ handle_unmount (UDisksFilesystem      *filesystem,
                                          0,
                                          NULL);
 
-  if (!bd_fs_unmount (mount_point ? mount_point : udisks_block_get_device (block),
+  if (!bd_fs_unmount (udisks_block_get_device (block),
```

Comment:
Ah, right, I've completely forgot about the bind mounts case. After brief discussion with @vojtechtrefny we've agreed this is even more complicated. I'll ditch this commit and will deal with this separately - filed #712 instead.

I was even thinking about checking the `MountPoints` property for multiple entries and either unmount by mount point if multiple mounts or by device in case of a single mount. However that would be inherently racy the same way.